src/hooks/useAddToCart.ts

```diff
@@ -0,0 +1,21 @@
+import { enqueueSnackbar } from 'notistack';
+import { useAddLineItemMutation } from '../api/cartApi';
+import { isProductInCart } from '../utils/isProductInCart';
+import type { Cart } from '../types/cartApi';
+
+export const useAddToCart = (cart?: Cart) => {
+  const addItem = useAddLineItemMutation()[0];
```

Comment:
Maybe it would be good to return not only a function `addToCart` , but also the loading state, and then use it to disable Add-button.

```suggestion
  const [addItem, { isLoading }] = useAddLineItemMutation();
```

Reply:
Added